trieste/acquisition/multiobjective/analytic.py

```diff
@@ -0,0 +1,161 @@
+from itertools import product
+from math import inf
+from typing import Callable, Union
+
+import tensorflow as tf
+from tensorflow_probability import distributions as tfd
+
+from ...data import Dataset
+from ...models import ProbabilisticModel
+from ...type import TensorType
+from ...utils.pareto import Pareto
+from ..function import AcquisitionFunction
+from .function import HypervolumeAcquisitionBuilder, get_reference_point
+
+
+class Expected_Hypervolume_Improvement(HypervolumeAcquisitionBuilder):
+    """
+    Builder for the :func:`hv_probability_of_improvement` acquisition function
+    refer yang2019efficient
+    """
+
+    def __init__(self, ref_pt_calc_method: Union[str, Callable] = "default"):
+        """
+        :param ref_pt_calc_method the method of calculating the reference point,
+         either default or a callable
+        """
+        self._ref_pt_calc_method = ref_pt_calc_method
+
+    def __repr__(self) -> str:
+        return f"Expected Hypervolume Improvement({self._ref_pt_calc_method!r})"
+
+    def _calculate_ref_pt(self, pareto: Pareto, ref_pt_calc_method="default"):
+        """
+        calculate the reference point for hypervolme calculation
+        :param pareto: Pareto class
+        :param ref_pt_calc_method
+        """
+        if ref_pt_calc_method == "default":
+            return get_reference_point(pareto.front)
+        else:
+            assert callable(ref_pt_calc_method), ValueError(
+                "ref_pt_calc_method: {} do not understood".format(ref_pt_calc_method)
+            )
+            return ref_pt_calc_method(pareto.front)
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ProbabilisticModel
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: The data from the observer. Must be populated.
+        :param model: The model over the specified ``dataset``.
+        :return: The expecyed_hv_of_improvement function.
+        """
+        tf.debugging.assert_positive(len(dataset), message="Dataset must be populated.")
+        mean, _ = model.predict(dataset.query_points)
+
+        _pf = Pareto(mean)
+        _reference_pt = self._calculate_ref_pt(_pf, ref_pt_calc_method=self._ref_pt_calc_method)
+        return lambda at: self._acquisition_function(model, at, _pf, _reference_pt)
+
+    @staticmethod
+    def _acquisition_function(
+        model: ProbabilisticModel,
+        at: TensorType,
+        pareto: Pareto,
+        ref_pt: TensorType,
+    ) -> tf.Tensor:
+        return expected_hv_improvement(model, at, pareto, ref_pt)
```

Comment:
```suggestion
        return lambda at: expected_hv_improvement(model, at, _pf, _reference_pt)
```
i know we do this elsewhere, but we never actually use `_acquisition_function`. Meanwhile, I think it makes the code harder to read. What do you think?